### PR TITLE
Clarify change in behavior for druid.server.maxSize

### PR DESCRIPTION
**Pull request by a2l007**
With #10070 , `druid.server.maxSize` is no longer just a soft limit on the total segment size that can be assigned to a historical. If this property is set to the default value of 0 for a historical, coordinator will no longer assign segments to it. 
This doc change attempts to clarify this behavior.
Long term, it would be useful to identify a more realistic default value for this property instead of zero.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1370,7 +1370,7 @@ These Historical configurations can be defined in the `historical/runtime.proper
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`druid.server.maxSize`|The maximum number of bytes-worth of segments that the process wants assigned to it. This is not a limit that Historical processes actually enforces, just a value published to the Coordinator process so it can plan accordingly.|0|
+|`druid.server.maxSize`|The maximum number of bytes-worth of segments that the process wants assigned to it. The Coordinator process will attempt to assign segments to a Historical process only if this property is greater than the total size of segments served by it. Since this property defines the upper limit on the total segment size that can be assigned to a Historical, it needs to be set to a value greater than zero.|0|
```
Comment (clintropolis) [inline docs/configuration/index.md:1373]: Should this mention that this size should be equal(ish) to the sum of all segment cache location sizes? Ideally someday we can drop this property and just compute it from the segment cache config, but .. that day is not today, so maybe it is worth including.

**Comment by gianm** [comment]
What was the behavior change that happened in #10070?

I looked at that patch, and I read the diff in the docs here, and it isn't clear to me what changed.

**Comment by a2l007** [comment]
@gianm Prior to that patch, coordinator could assign segments to historicals even if the balancer couldn't find a valid move for a segment. This means that even if `druid.server.maxSize` isn't optimally configured, the coordinator could still assign segments to it.
However, with the patch the coordinator avoids segment assignment if it cannot find valid moves. This means that existing clusters configured with incorrect `druid.server.maxSize` property could see coordinators not assigning segments even though the historicals have enough capacity. So I feel the docs needs to emphasize the importance of this property a bit more given the behavioral change.

**Comment by gianm** [comment]
> @gianm Prior to that patch, coordinator could assign segments to historicals even if the balancer couldn't find a valid move for a segment. This means that even if `druid.server.maxSize` isn't optimally configured for a historical, the coordinator could still assign segments to it.

What do you mean by "could assign segments even if the balancer couldn't find a valid move"? Like, if all historicals were full (no room in `druid.server.maxSize`), it would assign to one of the full ones anyway?

**Comment by a2l007** [comment]
> > @gianm Prior to that patch, coordinator could assign segments to historicals even if the balancer couldn't find a valid move for a segment. This means that even if `druid.server.maxSize` isn't optimally configured for a historical, the coordinator could still assign segments to it.
> 
> What do you mean by "could assign segments even if the balancer couldn't find a valid move"? Like, if all historicals were full (no room in `druid.server.maxSize`), it would assign to one of the full ones anyway?

Yes I was referring to that behavior.

**Comment by gianm** [comment]
I see. That older behavior sounds like a bug that got fixed, but perhaps people were relying on the buggy behavior. So I agree it would be good to call it out. Thanks for explaining it.

**Pull request merged**